### PR TITLE
Empirical distributions quantiles

### LINKED ISSUE #1549: quantiles not implemented for Empirical distribution
Quantiles are not currently implemented for `tensorflow_probability.distributions.Empirical`
To reproduce: 

```python
import tensorflow_probability as tfp
samples = [1.,2.,3.,2.,1.]
dist = tfp.distributions.Empirical(samples=samples)
dist.quantile(0.5)

# raises: NotImplementedError
```

This should be an easy fix, using the already available in tensorflow_probability functions from the `tensorflow_probability.stats` module.

### PR DESCRIPTION
**Pull request by ltsaprounis**
Fixes #1549

Implemented the quantile method for the Empirical distribution class by interfacing the `tfp.stats.percentile` method and scaling the input appropriately. I think that the tests pass locally. I've tested using the command below: 
`bazel test -c opt tensorflow_probability/python/distributions:normal_test \
    --test_arg=--test_regex=EmpiricalScalarTest`

Let me know if I need to add more tests in the `EmpiricalNdTest` and the `EmpiricalVectorTest` classes. 

Thanks!

### CODE REVIEW

File: tensorflow_probability/python/distributions/empirical.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from tensorflow_probability.python.internal import samplers
 from tensorflow_probability.python.internal import tensor_util
 from tensorflow_probability.python.internal import tensorshape_util
-
+from tensorflow_probability.python.stats import percentile
```
Comment (SiegeLordEx) [inline tensorflow_probability/python/distributions/empirical.py:28]: Our style guide forbids us from importing functions directly, so this should just import stats and then later on the call should be `stats.percentile`

File: tensorflow_probability/python/distributions/empirical.py [hunk 2]
```diff
@@ -217,6 +217,18 @@ def _stddev(self):
     r = samples - tf.expand_dims(self._mean(samples), axis=axis)
     var = tf.reduce_mean(tf.square(r), axis=axis)
     return tf.sqrt(var)
+  
+  def _quantile(self, value, samples=None, **kwargs):
+    if value > 1 or value < 0:
```
Comment (SiegeLordEx) [inline tensorflow_probability/python/distributions/empirical.py:222]: You don't need to do this check here, it's already handled by the base Distribution class.

**Review by SiegeLordEx — CHANGES_REQUESTED** [review]
Thanks for the change. Sorry for the delay, some weird issue with the CLA bot not telling us that the CLA was signed. I left a few comments.

**Comment by ltsaprounis** [comment]
> Thanks for the change. Sorry for the delay, some weird issue with the CLA bot not telling us that the CLA was signed. I left a few comments.

Thanks for the review! I addressed all the requested changes and the [linting checks](https://github.com/tensorflow/probability/blob/main/CONTRIBUTING.md#helper-scripts)  from the CONTRIBUTING.md now pass locally.

**Review by SiegeLordEx — APPROVED** [review]

**Comment by SiegeLordEx** [comment]
Thanks! I'll send this along to our internal review system and we'll get this merged once that is done.

**Comment by ltsaprounis** [comment]
According to the [CI logs](https://github.com/tensorflow/probability/runs/6359160764?check_suite_focus=true#step:4:1860) the test I wrote for the quantiles `test_empirical_quantiles` is failing and the following error is raised: 
`AttributeError: module 'tensorflow_probability.python.internal.backend.numpy.v2' has no attribute 'bfloat16'`

@SiegeLordEx - Do you have any idea what this means? I'm a bit stuck on this...

**Comment by SiegeLordEx** [comment]
The percentile function is erroneously exposed by the numpy backend. What I'll do is just disable the test and the quantile implementation under numpy.

**Comment by ltsaprounis** [comment]
@SiegeLordEx - thanks for merging this! 

> The percentile function is erroneously exposed by the numpy backend. What I'll do is just disable the test and the quantile implementation under numpy.

Any idea why this happens? 
Apologies if this sounds naive, I'm not that familiar with the backend of tensorflow probability, but numpy has a percentile function of it's own, is there a way to resolve this by replacing the percentile function with the native numpy function?

**Comment by SiegeLordEx** [comment]
TFP has its own [percentile function implementation](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/stats/quantiles.py#L428) which supports batching and (outside of numpy, were it to work) gradients.

Currently the module where the TFP's percentile is defined is erroneously exposed under the numpy backend. This is because of a misconfiguration of the [bazel target](https://github.com/tensorflow/probability/blob/main/tensorflow_probability/python/stats/BUILD#L157), which should ideally indicate that it doesn't work under numpy. Unfortunately it's tricky to fix, since there's a bit of a cascading set of changes that would need to be made to change just that one target. We haven't had the time to do that.

**Pull request merged**